src/Components/Cart.jsx

```diff
@@ -1,8 +1,20 @@
-import { Dialog, Transition } from "@headlessui/react";
-import { XIcon } from "@heroicons/react/outline";
-import React, { Fragment } from "react";
+import { Dialog, Transition } from '@headlessui/react';
+import { XIcon } from '@heroicons/react/outline';
+import React, { Fragment, useEffect } from 'react';
+import { ShoppingCartIcon } from '@heroicons/react/outline';
 
 export default function Cart({ open, setOpen, cart, updateCart }) {
+  const displaySubtotal = () => {
+    let subtotal = 0;
+    cart.forEach((element) => {
+      // setCartSubTotal(cartsubtotal + element.price);
+      let oneprice = element.quantity * element.price;
+      subtotal += oneprice;
+    });
+
+    return subtotal;
+  };
+
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought:** The displaySubtotal() function calculates the subtotal by iterating through the cart array. Consider using a more functional approach with reduce:
```js
const displaySubtotal = () => {
  return cart.reduce((subtotal, element) => {
    return subtotal + element.quantity * element.price;
  }, 0);
};
```